lib/gui/pages/settings/controllers/settings.js

```diff
@@ -80,10 +86,37 @@ module.exports = function(WarningModalService, SettingsModel) {
 
     WarningModalService.display(options).then((userAccepted) => {
       if (userAccepted) {
+        AnalyticsService.logEvent('Set setting', {
+          setting: name,
+          value: true,
+          dangerous: true
+        });
+
         this.model.set(name, true);
         this.refreshSettings();
       }
     });
   };
 
+  /**
+   * @summary Log event and delegate to model.
+   * @function
+   * @public
+   *
+   * @description
+   * The logging is done here for now and not in the model to avoid
+   * a circular dependency
+   *
+   * @param {String} key - setting key
+   * @param {Object} value - setting value
+   */
+  this.set = (key, value) => {
+    AnalyticsService.logEvent('Set setting', {
+      setting: key,
+      value
```

Comment:
Again, apologies if I'm miles off the mark, but to match up with the other similar changes here should this be `value: value` ?